src/Aspire.Dashboard/Model/ThemeManager.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Dashboard.Model;
+
+public sealed class ThemeManager
+{
+    private readonly object _lock = new object();
+    private readonly List<ModelSubscription> _subscriptions = new List<ModelSubscription>();
+
+    // Note: This won't have a valid value until it has been changed.
+    // If there is a reason to get the theme before changes then the ThemeManager will need to be improved.
+    public string? Theme { get; private set; }
```

Comment:
Would be good if we can somehow combine this with theme setting change directly so theme manager becomes the object which anyone can ask to get current theme. rather than settings dialogue holding all that.

I am also fine to merge with this form and improve later.

Reply:
Sure, in the future. That's why the comment is here.